File: Day 9/day9.py
from typing import List, Literal

def move_train(board: List[str], mov: Literal['U', 'D', 'R', 'L']) -> Literal['none', 'crash', 'eat']:
  suma_mov = {'U': (-1,0),'D': (1,0), 'R': (0,1),'L': (0,-1)}
  resultado = {'*': 'eat', 'o':'crash','·':'none'}

  m_i, m_j = suma_mov[mov]
  for i,linea in enumerate(board):
    if '@' in linea:
      nx,ny = i+m_i, linea.index('@')+m_j

  if not (0 <= nx < len(board) and 0 <= ny < len(board[0])):
    return 'crash'

  return resultado[board[nx][ny]]

File: Day 9/day9.ts
type Move = 'U' | 'D' | 'R' | 'L';
type Result = 'none' | 'crash' | 'eat';

function moveTrain(board: string[], mov: Move): Result {
    const sumaMov: Record<Move, [number, number]> = {
        'U': [-1, 0],
        'D': [1, 0],
        'R': [0, 1],
        'L': [0, -1]
    };

    const resultado: Record<string, Result> = {
        '*': 'eat',
        'o': 'crash',
        '·': 'none'
    };

    const [m_i, m_j] = sumaMov[mov];
    let nx: number = -1, ny: number = -1;

    for(let i = 0; i < board.length; i++){
        const linea = board[i];

        if (linea.includes('@')){
            nx = i + m_i;
            ny = linea.indexOf('@') + m_j;
        }
    }

    if (!(0 <= nx && nx < board.length && 0 <= ny && ny < board[0].length)) {
        return 'crash';
    }

    return resultado[board[nx][ny]]
}